libraries/AP_BattMonitor/AP_BattMonitor_UAVCAN.h

```diff
@@ -51,6 +51,8 @@ class AP_BattMonitor_UAVCAN : public AP_BattMonitor_Backend
     static void handle_battery_info_aux_trampoline(AP_UAVCAN* ap_uavcan, uint8_t node_id, const BattInfoAuxCb &cb);
     static void handle_mppt_stream_trampoline(AP_UAVCAN* ap_uavcan, uint8_t node_id, const MpptStreamCb &cb);
 
+    static float calculate_mah(float amps, float dt) { return (float) (amps * dt * 0.0000002778f); }
```

Comment:
Needs a comment to say where the magic constant came from, and arguably should actually be a helper that is in the general battery backend (or a seperate static function that any of the backends could include as needed), since the analog monitor has the same code path and we should be trying to hold these to do identical calculations.

Reply:
Alright, I made it a static function in the backend header. Thought about virtual but it would make it harder to run the test class, although its currently causing errors for some reason it might just be best to scrap it.

I also made Analog use the new function.